components/learn/modules/ROOT/pages/public-staging.adoc

```diff
@@ -1,39 +1,181 @@
+We will use *public testing networks* (aka _testnets_) for this, which are networks that operate similar to the main Ethereum network, but where Ether has no value and is free to acquire - making them ideal for testing your contracts at no cost.
 
-difference from unit testing
-production-like environment
+In this guide, we will pick up our xref:deploy-and-interact.adoc#box-contract[beloved `Box` contract], and deploy it to a testnet, while learning:
 
-== setting up
+* <<testnet-list, What test networks are available>>
+* <<connecting-project-to-network, How to set up your OpenZeppelin project for working on a testnet>>
+* <<deploy-and-interact, How to deploy and interact with your testnet contract instances>>
 
-choose a network (probably ropsten or rinkeby, ropsten is closer to mainnet but rinkeby is more reliable)
+Remember that deploying to public test network is a necessary step when developing an Ethereum project. They provide a safe environment for testing that closely mimics the main network - and you don't want to take out your project for its test drive in an environment where mistakes will actually cost you money!
 
-=== connection
-infura (explain, get account)
+[[testnet-list]]
+== Available Testnets
 
-=== funds
-get funds (faucet, depends on network)
-hdwallet provider
+There are four test networks available for you to choose, each with their own characteristics:
 
-=== configure w/dotenv
-use dotenv and hdwallet provider
-add network entry to networks.js
+[horizontal]
+Ropsten:: The only proof-of-work testnet. It has unpredictable block times and frequent chain reorganizations. At the same time, it is the chain that most closely resembles mainnet. (id=3)
+Rinkeby:: A proof-of-authority network. This means that new blocks are added by a set of pre-defined trusted nodes, instead of by whichever miner provides a proof-of-work. It only works with Geth clients, and has 15-second block times. (id=4)
+Kovan:: Another proof-of-authority network, but this one runs only with Parity clients, and has 4-second block times. (id=42)
+Goerli:: Also a proof-of-authority network, but compatible with both Geth and Parity clients, with 15-second block times. (id=6)
 
-== send CLI commands
+NOTE: Each network is identified by a numeric ID. Development networks usually have a large random value, while id=1 is reserved for the main Ethereum network.
 
-same as local, but:
-  not instant
-  persistent forever in <network>.json files
-  `from` is important
+It is up to you whether you want to test in a more unpredictable environment, to assess how robust your application is, or a more reliable one, to simplify your testing experience.
 
-note that call is still free and almost instant
+[[connecting-project-to-network]]
+== Connecting a Project to a Testnet
 
-== programmatic CLI commands
+To connect our OpenZeppelin project to a public testnet, we will need to:
 
-same caveats apply, using loader still works
+  . Get hold of a testnet node
```

Comment:
Can (should) we make these links?

Reply:
Yes, but I was lazy. Making them links now!